src/Geta.GoogleProductFeed/Repositories/FeedRepository.cs

```
@@ -9,37 +9,57 @@ namespace Geta.GoogleProductFeed.Repositories
+        {
+            _applicationDbContext = applicationDbContext;
+        }
+
         public void RemoveOldVersion()
         {
-            var items = FeedStore.Items<FeedData>().OrderByDescending(f => f.Created).ToList();
-            if (items.Count > 1)
+            try
             {
-                for (int i = items.Count - 1; i >= 1; i--)
+                var items = _applicationDbContext.FeedData.OrderByDescending(f => f.CreatedUtc).ToList();
+                if (items.Count > 1)
                 {
-                    FeedStore.Delete(items[i].Id);
+                    for (int i = items.Count - 1; i >= 1; i--)
+                    {
+                        _applicationDbContext.FeedData.Remove(items[i]);
+                    }
+
+                    _applicationDbContext.SaveChanges();
                 }
             }
+            catch (Exception ex)
+            {
+                _log.Error("Error removing old Google Product Feed using Entity Framework", ex);
+            }
         }
 
         public FeedData GetLatestFeedData()
         {
-            return FeedStore.Items<FeedData>().OrderByDescending(f => f.Created).FirstOrDefault();
+            return _applicationDbContext.FeedData.OrderByDescending(f => f.CreatedUtc).FirstOrDefault();
         }
 
         public void Save(FeedData feedData)
         {
             if (feedData == null)
+            {
                 return;
+            }
 
-            FeedStore.Save(feedData);
-        }
-
+            feedData.CreatedUtc = DateTime.UtcNow;
 
-        private static DynamicDataStore FeedStore
-        {
-            get
+            try
+            {
+                _applicationDbContext.FeedData.Add(feedData);
+                _applicationDbContext.SaveChanges();
+            }
+            catch (Exception ex)
             {
-                return typeof(FeedData).GetStore();
+                _log.Error("Error saving Google Product Feed using entity framework", ex);
```

It should not be up to the repository to decide how to handle saving errors. Now it just blindly swallows any exceptions, caller never knows if saving failed or not.